src/proxy-builder.test.ts

```diff
@@ -12,22 +12,29 @@ describe("Test configureProxy", () => {
     process.env.HTTPS_PROXY = https_env;
     const result = configureProxy("https://test.com:8000");
 
-    expect(result).toEqual({
-      proxy: {
-        hostname: "testproxy.com",
-        port: 8000,
-        protocol: "https:",
-        auth: {
-          username: "",
-          password: "",
-        },
+    expect(result.proxy).toEqual({
+      hostname: "testproxy.com",
+      port: 8000,
+      protocol: "https:",
+      auth: {
+        username: "",
+        password: "",
       },
-    } as RequestProxy);
+    });
+    expect(result.httpsAgent).toBeUndefined();
+  });
+
+  test("with tunnel agent - with http env in proxy", () => {
+    process.env.HTTPS_PROXY = http_env;
```

Comment:
as you have added the fallback now, you can use

process.env.HTTP_PROXY = http_env;

Reply:
oh. seems like you removed the fallback. okay. we need to keep as it is

Reply:
I don't believe so, with the fallback reverted to the way `request` did it? It fails tests.

Reply:
that's weird

Reply:
if you have fallback in place, then process.env.HTTP_PROXY = http_env shouldn't cause the issue 

anyway, let's go with this for now. we can cover it in another PR. i can give a try